### DISCUSSION TITLE
Where's Socrates?

### DISCUSSION THREAD

**Discussion by michaeljohnston0** [EvgSkv/logica#296] [General]
Brand new to Logica.  Went through a couple tutorials.  Excited...hopeful...

My heart breaks for the lack of a "Socrates is a man, all men are mortal, therefore Socrates is mortal" example.  Moreover, for logic programming, I'm used to a question prompt, for code like:

```
man(socrates).
mortal(X) :- man(X).
```

And then being able to "ask": 

```
?mortal(socrates).
```

Any tips?  Happy to "RTFM" but I do feel something's missing here...  

Moreover, this might be worth promoting more for the project.  All the tutorials I've seen show me how to basically write SQL with Logica... if I knew exactly the SQL I wanted, if SQL was a convenient way to think of it, I wouldn't be looking for a language with "logic" in the name (and some others seem to also think Logica as a handy sql-writing tool isn't interesting enough, for example https://www.reddit.com/r/dataengineering/comments/n644qg/thought_about_logica_google_new_programming/).   I think with more emphasis on how to do more open-ended logical deduction (if it does support that well) and perhaps graph analysis (the first application I'd like to try) could help Logica get off the ground more too.

It said "don't hesitate to ask questions"...

**Comment by yilinxia** [comment]
Hi @michaeljohnston0 , you can try the following in [playgroud](https://logica.dev/sandbox.html?program=Man%28%22Socrates%22%29%3B+%0AMortal%28x%29+%3A-+Man%28x%29%3B&main_predicate=Mortal)
```
Man("Socrates"); 
Mortal(x) :- Man(x);
```
As far as I'm aware, Logica doesn't currently support the "ask" feature.

**Comment by EvgSkv** [comment]
@yilinxia thanks for answering! 😄 

@michaeljohnston0 Let us know if you have any followup questions on this.

Also you are welcome to start a thread about your usecase. I am interested to learn what you mean by 'open-ended' deduction. It may or may not be within Logica scope. If we think of graph problems: Distances in a graph, or the number of connected components are natural to declare in Logica. If you need to find whether graph is 3-colorable then SAT-solver-based solutions would probably be more practical than Logica.
